src/Core/Framework/Test/Plugin/Command/PluginListCommandTest.php

```diff
@@ -0,0 +1,164 @@
+                $plugin = $entities[$number];
+                $plugin->method($method)->willReturn($value);
+            }
+        }
+
+        $this->setupEntityCollection($entities);
+
+        [$statusCode, $output] = $this->executeCommand([]);
+        static::assertSame(0, $statusCode);
+        static::assertStringEqualsFile(
+            __DIR__ . '/../_assertion/PluginListCommandTest::testCommand.txt',
+            implode("\n", array_map('trim', explode("\n", $output))) . "\n"
+        );
+    }
+
+    public function testFilter(): void
+    {
+        $filterValue = 'shopware-is-love';
+
+        $criteria = static::callback(function (Criteria $criteria) use ($filterValue): bool {
+            $filters = $criteria->getFilters();
+            // must be MultiFilter
+            if (!(\count($filters) === 1 && $filters[0] instanceof MultiFilter)) {
+                return false;
+            }
+            /** @var MultiFilter $filter */
+            $filter = $filters[0];
+            // must be OR
+            if ($filter->getOperator() !== MultiFilter::CONNECTION_OR) {
+                return false;
+            }
+            $fields = ['name', 'label'];
+            foreach ($filter->getQueries() as $query) {
+                /** @var ContainsFilter $query */
+                if (!(
+                    $query instanceof ContainsFilter
+                    && $query->getValue() === $filterValue
+                    // first test against name, then label
+                    && $query->getField() === array_shift($fields)
+                )
+                ) {
+                    return false;
+                }
+            }
+
+            return true;
+        });
+
+        $this->pluginRepoMock->method('search')->with($criteria, static::anything());
+
+        [$statusCode, $output] = $this->executeCommand(['--filter' => $filterValue]);
+        static::assertSame(0, $statusCode);
+        static::assertStringContainsString('Filtering for: ' . $filterValue, $output);
+    }
+
+    public function testJsonOutput(): void
+    {
+        $o1 = ['PLUGIN1'];
+        $o2 = ['PLUGIN2'];
+
+        $entities = [
+            $plugin1 = $this->createMock(PluginEntity::class),
+            $plugin2 = $this->createMock(PluginEntity::class),
+        ];
+
+        $plugin1->method('jsonSerialize')->willReturn($o1);
+        $plugin2->method('jsonSerialize')->willReturn($o2);
+
+        $this->setupEntityCollection($entities);
+
+        $options = ['--json' => true];
+        $json = json_encode([$o1, $o2], \JSON_THROW_ON_ERROR);
+        $expected = [0, $json];
+        static::assertSame($expected, $this->executeCommand($options));
+    }
+
+    private function executeCommand(array $options): array
+    {
+        $commandTester = new CommandTester($this->command);
+        $commandTester->execute($options);
+
+        return [$commandTester->getStatusCode(), trim($commandTester->getDisplay())];
```

Comment:
i think it would be clearer to return the command tester instead, and then do two specific asserts on those methods

IMHO the wrapping in an array of the expected and actual values obscures the essence of the tests